src/store.rs

```diff
@@ -19,11 +19,11 @@ pub trait StoreParams: Clone + Send + Sync + Unpin + 'static {
 }
 
 /// Default store parameters.
-#[derive(Clone)]
-pub struct DefaultStoreParams;
+#[derive(Clone, Default)]
+pub struct DefaultParams;
 
-impl StoreParams for DefaultStoreParams {
-    const MAX_BLOCK_SIZE: usize = usize::MAX;
+impl StoreParams for DefaultParams {
+    const MAX_BLOCK_SIZE: usize = 1_048_576;
```

Comment:
@vmx doesn't like this. To some degree it's a parameter of the store, and the MemStore limit is usize

Reply:
Can you please revert it? You're aware that it's a contentious issue

Reply:
I don't see how this change prevents the use case @vmx was going for. A user which does want a different block size limit can create an own struct to impl StoreParams on. I think something called DefaultParams is likely to be used without second thought, and it seems prudent to not support an unreasonable amount of variability in the default case. I certainly didn't expect from cursory reading of the API docs that that limit might differ between 32bit and 64bit machines. Of course, maybe we're a 16bit machine, but at least compilation will tell you about too large literals. Silently compiling a different size however is a bad idea.

Reply:
I actually agree and don't object to the change for technical reasons. But you know from previous discussions that @vmx does not, and @vmx and I found a compromise. I don't think sneaking changes that you know are contentious are a good way of collaborating on a project.

Reply:
I was not aware this was contentious, since I believed the use case of @vmx to be covered. I'll revert the change, but hope to hear from @vmx as to whether it can be made amicably.

Reply:
@troublescooter thanks for reverting, @dvc94ch thanks for pointing out.

> I think something called DefaultParams is likely to be used without second thought, and it seems prudent to not support an unreasonable amount of variability in the default case.

That's the friction we have. As far as I understand you see libipld in a peer to peer/blockchain context, where such a default limit totally makes sense. I see libipld in a broader context, e.g. I might even store large raw data blocks locally and want to be able to link to them. So for me "default" means "the user can store whatever they like and things don't break".

Reply:
I'm not sure I can follow. I assumed the StoreParams construction to have been made to accomodate different blocksizes. If your use case is not covered by configuring your store manually, and having all possible stores have a block size limit specified by the machine, what is the configurability of the store for, @dvc94ch ?
If you don't provide a manually configured block bound, at the latest at runtime there will be a crash if a block bigger than `usize::MAX` is deserialised. This cannot be made to just work. My proposed fix catches this at compilation instead of runtime.

Reply:
I'm currently rewriting bitswap to be more robust. The maximum block size will be configurable, so anything below usize::MAX will work. If that's sensible or not is a completely different question. @vmx bitswap will no longer be compatible with pl implementations.